agent/runtime-attach/src/main/java/com/microsoft/applicationinsights/attach/ApplicationInsights.java

```diff
@@ -100,6 +119,23 @@ private static InputStream findResourceAsStream(String fileName) {
     return configContentAsInputStream;
   }
 
+  @Nullable
+  private static InputStream findJsonConfigFromFileSystemAsStream() {
+    File defaultFile = new File("config/applicationinsights.json");
+    if (!defaultFile.exists()) {
+      defaultFile = new File("applicationinsights.json");
+    }
+    if (!defaultFile.exists()) {
+      return null;
+    }
+    try {
+      return Files.newInputStream(defaultFile.toPath());
+    } catch (IOException e) {
+      throw new IllegalStateException(
```

Comment:
A `ConfigurationException` may be better: https://github.com/microsoft/ApplicationInsights-Java/blob/b2ecd8e74cf190d43cfac4ad8056337b86ce359c/agent/runtime-attach/src/main/java/com/microsoft/applicationinsights/attach/ApplicationInsights.java#L139C23-L139C46